neuro_config_client/factories.py

```diff
@@ -814,3 +830,85 @@ def create_ingress(cls, ingress: IngressConfig) -> dict[str, Any]:
         if ingress.cors_origins:
             result["cors_origins"] = ingress.cors_origins
         return result
+
+    @classmethod
+    def create_node_pool(
+        cls, node_pool: NodePool, cloud_provider_type: CloudProviderType
+    ) -> dict[str, Any]:
+        if cloud_provider_type == CloudProviderType.ON_PREM:
+            return cls._create_onprem_node_pool(node_pool)
+        elif cloud_provider_type.value.startswith("vcd_"):
+            return cls._create_vcd_node_pool(node_pool)
+        else:
+            return cls._create_public_cloud_node_pool(node_pool, cloud_provider_type)
+
+    @staticmethod
+    def _create_onprem_node_pool(node_pool: NodePool) -> dict[str, Any]:
+        result = {
+            "name": node_pool.name,
+            "role": node_pool.role.value,
+            "min_size": node_pool.min_size,
+            "max_size": node_pool.max_size,
+            "cpu": node_pool.cpu,
+            "available_cpu": node_pool.available_cpu,
+            "memory_mb": node_pool.memory_mb,
+            "available_memory_mb": node_pool.available_memory_mb,
+            "disk_size_gb": node_pool.disk_size_gb,
+            "price": str(node_pool.price),
+        }
+        if node_pool.disk_type:
+            result["disk_type"] = node_pool.disk_type
+        if node_pool.gpu:
+            result["gpu"] = node_pool.gpu
+        if node_pool.gpu_model:
+            result["gpu_model"] = node_pool.gpu_model
+        if node_pool.currency:
+            result["currency"] = node_pool.currency
+        return result
+
+    @staticmethod
+    def _create_public_cloud_node_pool(
+        node_pool: NodePool, cloud_provider_type: CloudProviderType
+    ) -> dict[str, Any]:
+        result: dict[str, Any] = {
+            "id": node_pool.id,
+            "name": node_pool.name,
+            "role": node_pool.role.value,
+            "min_size": node_pool.min_size,
+            "max_size": node_pool.max_size,
+            "disk_size_gb": node_pool.disk_size_gb,
+        }
+        if node_pool.idle_size:
+            result["idle_size"] = node_pool.idle_size
+        if node_pool.is_preemptible:
+            result["is_preemptible"] = node_pool.is_preemptible
+        if node_pool.disk_type:
+            result["disk_type"] = node_pool.disk_type
+
+        if cloud_provider_type == CloudProviderType.AWS:
```

Comment:
What if developer passes node_pool.gpu / gpu_model for cpu node? I think he should receive HTTP 400 error from server, but in your case he receives HTTP 200.

Reply:
Do you mean, we should pass zones, gpu and gpu_model info if they are present for all cases?

For me, as from an end-user perspective, this is mostly the same as we also pass cpu/memory info.
It's not clear which info should be included in the node pool description, especially since the HTTP error response from the server does not include info regarding what is wrong in the node pool description (validation results are not included).

I could include zones, gpu, gpu_model to all three types, but we should also send back the validation results in case of error. WDYT?

Reply:
Yes, pass all the data. Validation results is another problem, it needs to be tackled separately for all the client requests.

Reply:
Done in d229e1f